### PR TITLE
[WS-2984] Update architecture documentation for CMS

### PR DESCRIPTION
**Pull request by tomaszmichalak**
## Context

The content publishing pull model architecture docs and diagrams update.

### CODE REVIEW

File: docs/cms/architecture/index.md [hunk 1]
```diff
@@ -1,61 +1,37 @@
 # Architecture
-The diagram below represents the WebSight CMS high-level logical architecture. 
+WebSight CMS is a `Java`-based [Apache Sling](https://sling.apache.org/) content-centric application built on top of a `Java Content Repository (JCR)`. It uses `Apache Jackrabbit Oak` as the JCR implementation and `MongoDB` as the Node Storage. The application runtime is based on the `Apache Felix (OSGi)` framework.
```
[the remark this answers is not in the document]
Comment (tomaszmichalak) [reply docs/cms/architecture/index.md:2]: Do we want to describe two configurations: with Mongo and TAR?
My understanding is that the recommended production setup is with Mongo. It follows the twelve-factor app principle: `any data that needs to persist must be stored in a stateful backing service, typically a database`.
Introducing TARs for production we would face issue such as scalability and reliability. And we have no good answer without additional replication mechanisms etc.

[the remark this answers is not in the document]
Comment (tomaszmichalak) [reply docs/cms/architecture/index.md:2]: I use notes to show the capability but it is not what we recommend for the staging/production environments.

File: docs/cms/architecture/index.md [hunk 2]
```diff
@@ -1,61 +1,37 @@
...
 
-![WebSight - logical architecture](logical-architecture.jpg)
+![WebSight CMS with MongoDB](websight-cms.png)
 
-**Content authors** use **CMS** to **create** an experience by managing content, assets and pages (layout). Once the experience is ready to go public, authors push (in other words, publish) it to the backend **experience storage**.
+!!! note "Node Storage"
+    Apache Oak supports multiple [storage backends](https://jackrabbit.apache.org/oak/docs/nodestore/overview.html), including MongoDB and Segment Tar. WebSight CMS uses MongoDB as the recommended Node Storage for production environments. For development the Segment Tar storage is used by default.
```
[the remark this answers is not in the document]
Comment (tomaszmichalak) [reply docs/cms/architecture/index.md:7]: I used this diagram as a reference: https://jackrabbit.apache.org/oak/docs/architecture/overview.html.
When we talk about the storage we understand it as the bottom level: Tar files, MongoDB and RDBMS. Node stores are more abstractions above. In the OAK documentation they also use the term: `node storage` (https://jackrabbit.apache.org/oak/docs/nodestore/overview.html). And then they use:
> The node storage implement the NodeStore APIs...

I tried to reference to the storage as it is a part of the high-level architecture. Node Stores are more OAK-specific APIs, which are defined on the diagram above as a separate box. WDYT?

File: docs/cms/architecture/index.md [hunk 3]
```diff
@@ -1,61 +1,37 @@
...
 
-Our CMS is a `Java`-based `OSGi` application (**WebSight CMS CE**) with a `NoSQL` database (**MongoDB**). WebSight CMS supports content management and the generation of static HTML pages. MongoDB database stores content and assets.
+![WebSight CMS project](websight-cms-project.png)
 
-![WebSight - logical architecture with MongoDB](logical-architecture-detailed.jpg)
+Read our ["Building OSGi applications with Sling Feature Model"](https://www.websight.io/blog/2022/building-osgi-applications-with-sling-feature-model.html) blog post to learn more about the Sling Feature Model.
 
-!!! Info "Note"
-        Multiple instances of WebSight CMS can connect to MongoDB, which supports online deployments and enables CMS scalability.
+!!! note "Node Storage"
```
[the remark this answers is not in the document]
Comment (tomaszmichalak) [reply docs/cms/architecture/index.md:19]: I described it above, the link is in the Oak documentation: https://jackrabbit.apache.org/oak/docs/nodestore/overview.html

File: docs/cms/architecture/index.md [hunk 4]
```diff
@@ -1,61 +1,37 @@
...
-
-## Environments
-
-Working with containers enables consistency and standardization. With Docker, we can use the same tooling to work locally as well as to deploy production environments to the cloud.
-
-For example:
-
-When using Docker Compose, the same command is responsible for creating local and cloud instances (`docker compose up`). 
-
-Read more about [Docker Compose Amazon Elastic Container Service integration](./aws-ecs/).
```
[the remark this answers is not in the document]
Comment (tomaszmichalak) [reply docs/cms/architecture/index.md]: Agreed to reuse some parts and extend with the production setup references (Kubernetes and Helm).

**Comment by tomaszmichalak** [comment]
@michalcukierman The architecture docs are ready for merging. In the next PRs we plan to clean parts connected with ECS and Kubernetes.

**Review by paweljozwicki — APPROVED** [review]

**Pull request merged**